src/models/subscription.ts

```diff
@@ -72,6 +72,41 @@ export default class Subscription extends Sequelize.Model {
 		});
 	}
 
+	static getAllFiltered(
+		installationIds: number[],
+		statusTypes: string[],
+		limit: number,
+		offset: number
+	): Promise<Subscription[]> {
+
+		const andFilter = [];
+
+		if (statusTypes && statusTypes.length > 0) {
```

Comment:
can use optional chaining `statusTypes?.length`.  Also, if statusTypes might be undefined, you need to set the parameter to optional by adding `?` at the end of the name.  I think the end result should be something like:

```
statusTypes: string[] = ["FAILED", "PENDING", "ACTIVE"],
installationIds: number[] = [],
offset: number = 0,
limit?: number,
```
This way, you don't even need the 2 conditional to check for length.

Reply:
I moved the defaults to the model and simplified the if-conditions, but we can't remove the conditional checks, because they're building the SQL query with an IN operator and if the list we pass into the IN operator is empty, we get an empty result